### ISSUE TITLE
how to replace json values ?

### ISSUE THREAD

**Issue by JaveleyQAQ** [PortSwigger/BChecks#13]
I want to loop payload into all the JSON values

**Comment by genuinemoses** [comment]
Hi @JaveleyQAQ, maybe you're looking for something similar to the following:

run for each:
  payload = "PAYLOAD_1", "PAYLOAD_2"

given request then
  send request
    replacing body: `{"key_1": "{payload}", "key_2": "{payload}"}`
  
Hope this helps!

**Comment by JaveleyQAQ** [comment]
> 

Yes. But I think I don't know the key names and quantities for JSON. I found the loop method, given query or body insertion point, but it seems to have failed. It matched the cookie parameters, sent extra requests, and did not correctly insert my payload.
![image](https://github.com/PortSwigger/BChecks/assets/132129852/d8f1400b-b55a-40d1-b32e-bf1f36edf458)

**Comment by olliewuk** [comment]
Have you tried changing the `given query or body insertion point` to either

- `given query or body or cookie insertion point` (as it looks like you want your payload in a cookie?)
or
- `given request` per this example? https://github.com/PortSwigger/BChecks/blob/main/examples/server-side-prototype-pollution.bcheck

**Comment by JaveleyQAQ** [comment]
thank your rep@olliewuk   Here is another question:
```
defind:
    a = `{random_str(5)}`

if {latest.response} matches "`{a}`" then
```


, why does it fail to match? What should I do to make it match successfully?

**Comment by JaveleyQAQ** [comment]
> Have you tried changing the `given query or body insertion point` to either您是否尝试将 `given query or body insertion point` 更改为
> 
> * `given query or body or cookie insertion point` (as it looks like you want your payload in a cookie?) `given query or body or cookie insertion point` （因为看起来你想在cookie中添加有效负载？）
>   or  或
> * `given request` per this example? https://github.com/PortSwigger/BChecks/blob/main/examples/server-side-prototype-pollution.bcheck `given request` 这个例子？https://github.com/PortSwigger/BChecks/blob/main/examples/server-side-prototype-pollution.bcheck

I don't want to insert payloads in the cookies. My code is using "given query or body insertion point", but it's scanning the cookie parameters and sending a lot of useless requests even though it's not inserting any payload.

**Comment by genuinemoses** [comment]
> thank your rep@olliewuk Here is another question:
> 
> ```
> defind:
>     a = `{random_str(5)}`
> 
> if {latest.response} matches "`{a}`" then
> ```
> 
> , why does it fail to match? What should I do to make it match successfully?

I think there are typos and formatting issues, try with the following code block:

```
define:
    a = {random_str(5)}

if {latest.response} matches "{a}" then
```

**Comment by A-J-C** [comment]
@JaveleyQAQ unfortunately we don't currently support variables within regex matching. We can note it as a feature request if you would like this supported? 

Instead, you could use `if {a} in {latest.response}` as the `in` keyword does support variables. 

As to better json support, we also have only basic full body insertion point currently available to BChecks, with potential regex replace rules being used to work around it (as seen in the server side pollution check Ollie shared). If you would like this feature we could also note it?

**Comment by JaveleyQAQ** [comment]
@A-J-C @olliewuk  @genuinemoses   Thank you very much for your collaboration!  https://github.com/smxiazi/xia_sql/blob/main/BurpExtender.java  I want to implement a plugin with a  bc passive  script, but I'm having trouble with JSON matching .  I think I need to spend some time understanding examples of server side pollution.

**Comment by abdilahrf** [comment]
I think  now its not possible to replace JSON Payload automatically.

**Comment by Michelle-PortSwigger** [comment]
That's not currently possible. We'll add your vote for this feature.

**Comment by JaveleyQAQ** [comment]
> That's not currently possible. We'll add your vote for this feature.这目前是不可能的。我们将添加您对此功能的投票。

Where will the vote be held?

**Comment by JaveleyQAQ** [comment]
> That's not currently possible. We'll add your vote for this feature.
Where will the vote be held?

**Comment by Michelle-PortSwigger** [comment]
Hi

You don't need to go anywhere to vote. When someone requests a new feature, we raise it here, and when additional people also express interest in it, we make a note of that, too, so we can see how many people are interested in the feature to help us prioritize ideas. Your votes are already logged against the feature request. We've done that for you. Sorry for any confusion!

**Comment by JaveleyQAQ** [comment]
> Hi 
> 
> You don't need to go anywhere to vote. When someone requests a new feature, we raise it here, and when additional people also express interest in it, we make a note of that, too, so we can see how many people are interested in the feature to help us prioritize ideas. Your votes are already logged against the feature request. We've done that for you. Sorry for any confusion!


Thanks, very much looking forward to this feature!
